### PR TITLE
Avoid rebuilding the world when files containing primitives change.

### PR DESCRIPTION
**Pull request by stedolan**
New and improved* version of #8690.

*: less broken

### CODE REVIEW

**Comment by trefis** [comment]
My intuition is that the dune build is going to be broken by this.
(It might be fine, I haven't checked. Do you mind checking?)

**Comment by gasche** [comment]
I ran a `precheck` [job](https://ci.inria.fr/ocaml/job/precheck/267/) on the INRIA CI -- which would catch the kind of issues it caught previously.

**Comment by stedolan** [comment]
This shouldn't break the dune build: this patch doesn't change the `gen_primitives.sh` script that both `make` and `dune` use. Make redirects the script's output, but this won't affect dune's use of it.

**Comment by trefis** [comment]
Awesome, thanks for checking!

**Review by gasche — APPROVED** [review]
The precheck job seems very happy with the change, mingw32 isn't finished yet and the rest is all passing.

I'll merge when the CI has completed.

File: Changes [hunk 1]
```diff
@@ -54,6 +54,9 @@ Working version
   revise generation of config/util.ml to better quote special characters
   (Xavier Leroy, review by David Allsopp)
 
+- #8690: avoid rebuilding the world when files containing primitives change.
+  (Stephen Dolan, review by ??)
```
Comment (gasche) [inline Changes]: Sebastien did the work of triaging the CI reports and could be mentioned here.

**Comment by gasche** [comment]
@stedolan either you pushed to the wrong branch, or that's a really sneaky technique to get your code in.

**Comment by stedolan** [comment]
Arrggh, my evil plan is foiled. (pushed to the right branch this time)

**Pull request merged**